Code/RDGeneral/Dict.h

```diff
@@ -76,9 +76,9 @@ class RDKIT_RDGENERAL_EXPORT Dict {
       for (size_t i = 0; i < other._data.size(); ++i) {
         const Pair &pair = other._data[i];
         Pair *target = nullptr;
-        for (size_t i = 0; i < _data.size(); ++i) {
-          if (_data[i].key == pair.key) {
-            target = &_data[i];
+        for (size_t j = 0; j < _data.size(); ++j) {
```

Comment:
I think it might be better to replace the whole double loop with more modern code.
What about something like this?
```
      for (const auto &opair : other._data) {
        Pair *target = nullptr;
        for (auto &dpair : _data) {
          if (dpair.key == opair.key) {
            target = &dpair;
            break;
          }
        }

        if (!target) {
          // need to create blank entry and copy
          _data.push_back(Pair(opair.key));
          copy_rdvalue(_data.back().val, opair.val);
        } else {
          // just copy
          copy_rdvalue(target->val, opair.val);
        }
      }

```

Reply:
Even better! Pushed your suggestion as such.